src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1179,8 +1179,7 @@ Namespace Microsoft.CodeAnalysis.Operations
                                                      statementInfo.MoveNextMethod,
                                                      isAsynchronous:=False,
                                                      boundForEachStatement.EnumeratorInfo.NeedToDispose,
-                                                     knownToImplementIDisposable:=boundForEachStatement.EnumeratorInfo.NeedToDispose AndAlso
```

Comment:
>boundForEachStatement.EnumeratorInfo.NeedToDispose [](start = 82, length = 50)

I am not comfortable with us trying to infer this value later on. Code that calculates it is complicated enough and I would prefer us to not take any chances.